front/components/agent_builder/AgentBuilderFormContext.tsx

```diff
@@ -176,8 +208,20 @@ export const agentBuilderFormSchema = z.object({
   actions: z.array(actionSchema),
 });
 
+export const agentBuilderMCPFormSchema = z.object({
+  agentSettings: agentSettingsSchema,
+  instructions: z.string().min(1, "Instructions are required"),
+  generationSettings: generationSettingsSchema,
+  actions: z.array(mcpActionSchema),
+});
```

Comment:
Why not reusing `agentBuilderFormSchema` and a union on actions? Something like:
```typescript
export const agentBuilderFormSchema = z.object({
  agentSettings: agentSettingsSchema,
  instructions: z.string().min(1, "Instructions are required"),
  generationSettings: generationSettingsSchema,
  actions: z.union([z.array(actionSchema), z.array(mcpActionSchema)]),
});

Reply:
Hm I thought type wouldn't work in some places but seems like it's okay, I will do that 👍